src/Avalonia.Controls/Utils/GridLayout.cs

```diff
@@ -0,0 +1,694 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Avalonia.Layout;
+using JetBrains.Annotations;
+
+namespace Avalonia.Controls.Utils
+{
+    /// <summary>
+    /// Contains algorithms that can help to measure and arrange a Grid.
+    /// </summary>
+    internal class GridLayout
+    {
+        /// <summary>
+        /// Initialize a new <see cref="GridLayout"/> instance from the column definitions.
+        /// <see cref="GridLayout"/> will forget that the layout data comes from the columns.
```

Comment:
> GridLayout will forget that the layout data comes from the columns.

I don't understand this statement.

Reply:
I mean that the `GridLayout` instance doesn't care about whether the definitions are rows or columns. It will not calculate the column or row differently.

Reply:
Ahhhh, yes that makes sense. Put what you said there in the comments ;)